srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java

```diff
@@ -0,0 +1,367 @@
+         * @param value
+         *            the value of the environment variable to add to {@link #buildEnvironment}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder buildEnvironmentVariable(String name, String value) {
+            this.buildEnvironment.put(name, value);
+            return this;
+        }
+
+        /**
+         * @param ioRedirects
+         *            see {@link BuildRequest#getIoRedirects()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder ioRedirects(IoRedirects ioRedirects) {
+            this.ioRedirects = ioRedirects;
+            return this;
+        }
+
+        /**
+         * @param projectRootDirectory
+         *            see {@link BuildRequest#getProjectRootDirectory()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder projectRootDirectory(Path projectRootDirectory) {
+            this.projectRootDirectory = projectRootDirectory;
+            return this;
+        }
+
+        /**
+         * @param scmUrl
+         *            a SCM url to add to {@link #scmUrls}. See {@link BuildRequest#getScmUrls()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder scmUrl(String scmUrl) {
+            this.scmUrls.add(scmUrl);
+            return this;
+        }
+
+        /**
+         * @param scmUrls
+         *            SCM urls to add to {@link #scmUrls}. See {@link BuildRequest#getScmUrls()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder scmUrls(Collection<String> scmUrls) {
+            this.scmUrls.addAll(scmUrls);
+            return this;
+        }
+
+        /**
+         * @param srcVersion
+         *            see {@link BuildRequest#getSrcVersion()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder srcVersion(SrcVersion srcVersion) {
+            this.srcVersion = srcVersion;
+            return this;
+        }
+
+        /**
+         * @param timeoutMs
+         *            see {@link BuildRequest#getTimeoutMs()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder timeoutMs(long timeoutMs) {
+            this.timeoutMs = timeoutMs;
+            return this;
+        }
+
+        /**
+         * @param verbosity
+         *            see {@link BuildRequest#getVerbosity()}
+         * @return this {@link BuildRequestBuilder}
+         */
+        public BuildRequestBuilder verbosity(Verbosity verbosity) {
+            this.verbosity = verbosity;
+            return this;
+        }
+    };
+
+    /**
+     * The verbosity level to use when configuring the {@link Builder}.
+     */
+    public enum Verbosity {
+        debug, default_("default"), quiet;
```

Comment:
Sounds as a nice trick, thanks for the idea :)

I must say, I took the three levels quite mechanically from what maven supports on command line ( `--debug`, nothing, `--quiet`). Thinking of it again, (at least for maven 3.4) it should actually be possible to configure the slf4j logger used by maven using the standard trace, debug, info, warn and error levels. In any case, I should add some JavaDoc to the individual levels.